apis/core.oam.dev/v1beta1/applicationrevision_types.go

```diff
@@ -50,6 +50,9 @@ type ApplicationRevisionSpec struct {
 	// +kubebuilder:validation:EmbeddedResource
 	// +kubebuilder:pruning:PreserveUnknownFields
 	ApplicationConfiguration runtime.RawExtension `json:"applicationConfiguration"`
+
+	// Resources records the finalized resources from Application.
+	Resources []common.RawComponent `json:"resources,omitempty"`
```

Comment:
The resources recorded here are duplicated with those recorded in other fields of `appRevision.Spec`. This new field makes it convenient to consume these raw resources. But may I ask if there's other usage or purpose of it?

IMO, these "finalized" resources are not truly finalized, they are not ready to emit to k8s cluster until they are processed (set metadata, set references, etc) by AppContext(AppConfig) currently. In this `Resources` field, different kinds (workloads,traits,policies) of resources are mixed together, which makes it hard to be processed by AppConfig logic.

Reply:
If the goal is to provide a convenient and flexible way to get finalized resources (ready to emit), maybe assemble pkg as we introduced in #1705  can help. It will play as a standalone lib, the only input is an AppRevision, the outcome is a bundle of ready-to-go resources, and they are categorized.

Reply:
Hi @captainroy-hy . Thanks for your review. Those are very good questions. Let me share my reasoning and thoughts below:

 > The resources recorded here are duplicated with those recorded in other fields of appRevision.Spec.  This new field makes it convenient to consume these raw resources. But may I ask if there's other usage or purpose of it?

Yes. The goal is to use rendered resources to replace those fields like components or appConfig. From controller's perspective, it would be **EASIER** to just emit or use those resources without the structure. This is very different from user/developers' perspective.

Thus, I envision that we would eventually deprecate the Component or AppConfig field in AppRev, and replace them with ready-to-use resources.

> If the goal is to provide a convenient and flexible way to get finalized resources (ready to emit), maybe assemble pkg as we introduced in #1705 can help. It will play as a standalone lib, the only input is an AppRevision, the outcome is a bundle of ready-to-go resources, and they are categorized.

That's fine. I agree with this for the current situation. But again, this seems like workaround to me. Think about how much work this has made users to do: 1. learn about Component and AppConfig in AppRev; 2. learn about the assemble pkg in order to emit them.

Thus, I don't think that makes any sense. The Component/Trait/Workflow/etc. concepts are UI objects. For infra level control logic, we probably don't need them

Reply:
Sure, I got it! Thx for the details.

And this is a crucial signal ... I should take care.
> Thus, I envision that we would eventually deprecate the Component or AppConfig field in AppRev, and replace them with ready-to-use resources.